src/features/donations/components/ShareOptions.jsx

```diff
@@ -8,20 +11,23 @@ import DownloadIcon from '../../../../public/assets/images/icons/share/Download'
 import DownloadSolid from '../../../../public/assets/images/icons/share/DownloadSolid';
 import InstagramIcon from '../../../../public/assets/images/icons/share/Instagram';
 import tenantConfig from '../../../../tenant.config';
-import ReactDOM from 'react-dom';
-import domtoimage from 'dom-to-image';
 import i18next from '../../../../i18n/';
 
 const { useTranslation } = i18next;
 const ShareOptions = (props) => {
-  const { t, i18n, ready } = useTranslation(['donate', 'common']);
+  const { t, ready } = useTranslation(['donate', 'common']);
   const config = tenantConfig();
 
   const titleToShare = ready ? t('donate:titleToShare') : '';
   const urlToShare = config.tenantURL;
   const linkToShare = config.tenantURL;
-  const userName = props.contactDetails.firstName + ' ' + props.contactDetails.lastName;
-  const textToShare = ready ? t('donate:textToShareLinkedin', { name: userName}) : '';
+  let textToShare = '';
+  // props.contactDetails may be undefined or empty for legacy donations or redeem
+  if (props.contactDetails && (props.contactDetails.firstName || props.contactDetails.lastName)) {
+    textToShare = ready ? t('donate:textToShareLinkedin', { name: `${props.contactDetails.firstName} ${props.contactDetails.lastName}` }) : '';
```

Comment:
if props.contactDetails.firstname is null or props.contactDetails.lastName is now, i think these may also crash as well.

Reply:
I assumed an undefined value would be parsed into an empty string, so there is only a space too much showing.